public/components/event_analytics/explorer/explorer.tsx

```diff
@@ -545,19 +545,57 @@ export const Explorer = ({
                         <EuiSpacer size="m" />
                       </>
                     )}
-                    <DataGrid
-                      http={http}
-                      pplService={pplService}
-                      rows={explorerData.jsonData}
-                      rowsAll={explorerData.jsonDataAll}
-                      explorerFields={explorerFields}
-                      timeStampField={queryRef.current![SELECTED_TIMESTAMP]}
-                      rawQuery={appBasedRef.current || queryRef.current![RAW_QUERY]}
-                      totalHits={_.sum(countDistribution.data?.['count()'])}
-                      requestParams={requestParams}
-                      startTime={appLogEvents ? startTime : dateRange[0]}
-                      endTime={appLogEvents ? endTime : dateRange[1]}
-                    />
+                    {isDefaultDataSourceType || appLogEvents ? (
+                      <>
+                        {countDistribution.data && countDistribution.data['count()'] ? (
+                          <DataGrid
```

Comment:
I think here can be optimized, datagrid is duplicated

Reply:
reduced redundancy here